### PR TITLE
Fix utils.py classification function errors

### PR DESCRIPTION
**Pull request by vinferrer**
Closes Nothing.

Changes proposed in this pull request:

- After merging with main the classification part of the code wasnt working
- Changed a few lines from classification function all back to normal

### CODE REVIEW

**Comment by vinferrer** [comment]
Guys i runned pytest and this"`feature_scores.txt` file isn't been created here is the error:
```
        # Make sure files are generated
        assert isfile(join(out_path, "classification_overview.txt"))
        assert isfile(join(out_path, "classified_motion_ICs.txt"))
        assert isfile(join(out_path, "denoised_func_data_nonaggr.nii.gz"))
>       assert isfile(join(out_path, "feature_scores.txt"))
E       AssertionError: assert False
E        +  where False = isfile('/home/vicente/nilearn_data/development_fmri/development_fmri/out/feature_scores.txt')
E        +    where '/home/vicente/nilearn_data/development_fmri/development_fmri/out/feature_scores.txt' = join('/home/vicente/nilearn_data/development_fmri/development_fmri/out', 'feature_scores.txt')

test_integration.py:51: AssertionError
-------------------------------------------------------------------------------------------- Captured log call --------------------------------------------------------------------------------------------
WARNING  aroma.aroma:aroma.py:50 Output directory /home/vicente/nilearn_data/development_fmri/development_fmri/out exists and will be overwritten.
================================================================================= 1 failed, 3 passed in 152.12s (0:02:32) =================================================================================
```
Did you forget to eliminate this line or this file is supposed to be generated and now it isn't?

**Comment by vinferrer** [comment]
@tsalo Could you take a look into this?

**Comment by eurunuela** [comment]
> Did you forget to eliminate this line or this file is supposed to be generated and now it isn't?

I've looked for the generation of that file and haven't found it. I guess we did not remove that assert. @tsalo may know better.

File: aroma/utils.py [hunk 1]
```diff
@@ -327,9 +327,8 @@ def classification(features_df, out_dir):
     )
 
     # Put the indices of motion-classified ICs in a text file (starting with 1)
-    motion_ICs = features_df[
-        "classification", features_df["classification"] == "rejected"
-    ].index.values
+    motion_ICs = features_df["classification"][features_df["classification"] == "rejected"].index
+    motion_ICs = motion_ICs.values
```
Comment (tsalo) [inline aroma/utils.py:331]: Why this change?

Comment (vinferrer) [reply aroma/utils.py:331]: That was failing too after correcting the `x.T` error

Comment (tsalo) [reply aroma/utils.py:331]: Oh, sorry I see it now. I didn't notice the `][` being changed to `, ` and misinterpreted it. I thought it was just a cosmetic change.

**Comment by tsalo** [comment]
The feature scores go to a `tsv` now, not a `txt`. If you change the extension, it should be fine. Here's the relevant lines:

https://github.com/Brainhack-Donostia/aroma/blob/59a21be69f37a3ce34fd77906998877c845287c0/aroma/utils.py#L303-L304

**Comment by vinferrer** [comment]
I am gonna modify the test_integration since previous changes make the `join(resources_path, "classification_overview.txt")` obsolete.
```
(Pdb) true_classification_overview.iloc[:, 1:]
    maximum RP correlation  Edge-fraction  High-frequency content  CSF-fraction
IC                                                                             
1                     0.65           0.65                    0.96          0.00
2                     0.86           0.66                    0.08          0.00
3                     0.89           0.62                    0.13          0.01
4                     0.62           0.80                    0.96          0.00
(Pdb) classification_overview.iloc[:, 1:]
    csf_fract  max_RP_corr       HFC classification
IC                                                 
0    0.004761     0.653423  0.962798       rejected
1    0.004964     0.867240  0.107143       rejected
2    0.005950     0.857130  0.094742       rejected
3    0.002666     0.608323  0.962798       rejected
```

**Comment by vinferrer** [comment]
Okay these changes should do it

**Comment by vinferrer** [comment]
@tsalo and @eurunuela. To sumarize. I had to change some files in the data folder since their structure changed in the lasts merges and changed the `test_integration.py` to make it work

**Comment by tsalo** [comment]
Can you post the original error that this fixes? It seems like the scope of the PR has increased and I want to make sure it's fixing the bug @eurunuela noticed in https://github.com/Brainhack-Donostia/aroma/pull/25#issuecomment-725899407.

**Comment by vinferrer** [comment]
> Can you post the original error that this fixes? It seems like the scope of the PR has increased and I want to make sure it's fixing the bug @eurunuela noticed in [#25 (comment)](https://github.com/Brainhack-Donostia/aroma/pull/25#issuecomment-725899407).

```
test_integration.py:31: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../aroma.py:243: in aroma_workflow
    motion_ICs = utils.classification(features_df, out_dir)
../utils.py:316: in classification
    proj = HYPERPLANE[0] + np.dot(x.T, HYPERPLANE[1:])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (array([[0.65333356, 0.86082873, 0.88353643, 0.61168498, 0.43822201,
        0.405519  , 0.52083742, 0.21823561, 0.790..., 0.4905189 , 0.47750809,
        0.47114375, 0.36552243, 0.4786795 , 0.57517467]]), array([ 9.95127548, 24.83331602]))
kwargs = {}
relevant_args = (array([[0.65333356, 0.86082873, 0.88353643, 0.61168498, 0.43822201,
        0.405519  , 0.52083742, 0.21823561, 0.790...05189 , 0.47750809,
        0.47114375, 0.36552243, 0.4786795 , 0.57517467]]), array([ 9.95127548, 24.83331602]), None)

>   ???
E   ValueError: shapes (2,44) and (2,) not aligned: 44 (dim 1) != 2 (dim 0)

<__array_function__ internals>:5: ValueError
-------------------------------------------------------------------------------------------- Captured log call --------------------------------------------------------------------------------------------
WARNING  aroma.aroma:aroma.py:50 Output directory /home/vicente/nilearn_data/development_fmri/development_fmri/out exists and will be overwritten.
```
Yeah i think is the same error, that's solve by changing x.T to x

**Comment by vinferrer** [comment]
We should finish this today. I got a few PR waiting for this

File: aroma/tests/data/classification_overview.txt [hunk 2]
```diff
@@ -1,45 +1,5 @@
...
-40	False	0.28	0.48	0.23	0.02
-41	False	0.45	0.47	0.23	0.01
-42	False	0.33	0.48	0.29	0.04
-43	False	0.29	0.37	0.16	0.04
-44	False	0.47	0.58	0.24	0.01
+IC	edge_fract	csf_fract	max_RP_corr	HFC	classification
+0	0.6673487194914882	0.004789119897316091	0.6239073998302727	0.9627976190476192	rejected
+1	0.6560402271978647	0.004793030026303001	0.8509440133229171	0.08234126984126983	rejected
+2	0.8037272625459998	0.0031916636290894155	0.6323438789815838	0.9627976190476192	rejected
+3	0.623519424401591	0.005318133138776824	0.8048700572409935	0.09474206349206347	rejected
```
Comment (eurunuela) [inline aroma/tests/data/classification_overview.txt]: Why does the new file only have 4 components?

And why does it start from 0?

Comment (vinferrer) [reply aroma/tests/data/classification_overview.txt]: This is literally the output. Could you pull the PR and try at your computer?

Comment (vinferrer) [reply aroma/tests/data/classification_overview.txt]: OHHHHHH, i think i know what's happening.

Comment (vinferrer) [reply aroma/tests/data/classification_overview.txt]: The result has more components. However you are only checking the first four: 
```
    true_classification_overview = pd.read_csv(join(resources_path, "classification_overview.txt"),
                                               sep="\t",
                                               index_col="IC",
                                               nrows=4,)
    classification_overview = pd.read_csv(
        join(out_path, "classification_overview.txt"), sep="\t", index_col="IC", nrows=4
    )
```
Look at the nrows=4 argument. So we could have all the components storaged in the repo but we are only using the 4 firsts for checking

Comment (eurunuela) [reply aroma/tests/data/classification_overview.txt]: We should have all of them saved. Then we get to choose how many we want to actually compare with.

Comment (vinferrer) [reply aroma/tests/data/classification_overview.txt]: Fair enough

Comment (vinferrer) [reply aroma/tests/data/classification_overview.txt]: let me update then

Comment (vinferrer) [reply aroma/tests/data/classification_overview.txt]: done

File: aroma/tests/test_integration.py [hunk 3]
```diff
@@ -54,26 +54,22 @@ def test_integration(skip_integration, nilearn_data):
...
+    assert np.allclose(true_classification_overview.iloc[:, :-1].values,
+                       classification_overview.iloc[:, :-1].values,
+                       atol=0.9)
 
     #  Check feature scores
-    f_scores = np.loadtxt(join(out_path, "feature_scores.txt"))
-    f_true = np.loadtxt(join(resources_path, "feature_scores.txt"))
-    assert np.allclose(f_true[0, :], f_scores[0, :], atol=0.01)
+    f_scores = pd.read_csv(join(out_path, "feature_scores.txt"), sep="\t")
+    f_true = pd.read_csv(join(resources_path, "feature_scores.txt"), sep="\t")
```
Comment (tsalo) [inline aroma/tests/test_integration.py]: You could use `read_table` instead and you wouldn't have to specify the separator.

File: aroma/utils.py [hunk 4]
```diff
@@ -301,7 +301,7 @@ def classification(features_df, out_dir):
                                 components identified as motion components
     """
     # Put the feature scores in a text file
-    features_df.to_csv(op.join(out_dir, "feature_scores.tsv"), sep="\t", index=False)
+    features_df.to_csv(op.join(out_dir, "feature_scores.txt"), sep="\t", index=False)
```
Comment (tsalo) [inline aroma/utils.py]: Why make this change? We want tsv files for BIDS Derivatives compatibility.

File: aroma/tests/test_integration.py (hunk 3 above)
Comment (vinferrer) [reply aroma/tests/test_integration.py:71]: ohhh you wanted me to do the change on the test script?

File: aroma/utils.py (hunk 4 above)
Comment (tsalo) [reply aroma/utils.py]: Sorry, I see that I said "change the extension" when this came up earlier. I just meant that you should update the _test_, not the file.

Comment (vinferrer) [reply aroma/utils.py]: ohhh you wanted me to do the change on the test script?

Comment (tsalo) [reply aroma/utils.py]: Yes, sorry. My original comment was poorly phrased.

File: aroma/tests/test_integration.py (hunk 3 above)
Comment (tsalo) [reply aroma/tests/test_integration.py]: This is a minor thing, though, and we can hold off on it until another PR.

File: aroma/utils.py (hunk 4 above)
Comment (vinferrer) [reply aroma/utils.py]: Sure i will make the change

**Comment by vinferrer** [comment]
The tsv change is done.

**Review by tsalo — APPROVED** [review]
LGTM. Just be aware that the fix here will probably conflict with #28.

**Comment by vinferrer** [comment]
Yes but we need the main branch to work properly
